### DISCUSSION TITLE
STM32G031G. I want to confirm that the 28 pin package can be programmed using pins 18 and 19 with bootloader

### DISCUSSION THREAD

**Discussion by JuanTopoSPA** [stm32duino/Arduino_Core_STM32#1873] [Q&A] [answered]
I understand that PA11 and PA12 are mapped by default, which don't have USART TX/RX function, it should be remapped to PA9 and PA10.
AN2606 says that P9 and P10 can be used, there's even a note that says: "_On SO8, WLCSP18, TSSOP20 and UFQFN28 packages USART1 PA9/PA10 IOs are remapped on PA11/PA12_.". Does it mean that the bootloader automatically remap those pins?

Cases where it was not achieved/ somebody answered that is not possible: 
https://github.com/stm32duino/Arduino_Core_STM32/issues/1637#issuecomment-1028954189   @ABOSTM 
https://bbs.21ic.com/forum.php?mod=viewthread&tid=3164398&page=1#pid12329378

and cases where they are using uart in pins that need to be remapped (although in packages with 8 pins)
https://community.st.com/s/question/0D53W0000056kdnSAA/using-uart-bootloader-on-stm32g030j6
https://github.com/stm32duino/Arduino_Core_STM32/issues/1637#issuecomment-1029093813   @olikraus 

(No, I can't use PA2/PA3 or I2C on PB6/PB7.. or SW pins 🌵)

**Comment by olikraus** [comment]
If the USART boot mode is enabled (boot selection bit must be cleared), then the boot loader will read from USART1, which are PA9 and PA10. I assume the boot loader will do any remapping if required.

**Comment by ABOSTM** [comment, marked as answer]
@JuanTopoSPA ,

> AN2606 says that P9 and P10 can be used, there's even a note that says: "On SO8, WLCSP18, TSSOP20 and UFQFN28 packages USART1 PA9/PA10 IOs are remapped on PA11/PA12.". Does it mean that the bootloader automatically remap those pins?

It means that on regular chip (where PA9/PA10 are available) you need to use PA9/PA10, no remap possible.
But on specific chip like SO8, WLCSP18, ...  where PA9/PA10 are not available, then bootloader remap USART1 to PA11/PA12.

In other words, PA11/PA12 are used by bootloader only when PA9/PA10 are not available on your package.
